dispatch.go

```diff
@@ -89,22 +90,35 @@ func newDispatch(enc encoder, dec byteReadingDecoder, l LogInterface, wef WrapEr
 
 type call struct {
 	ch             chan error
+	doneCh         chan struct{}
 	method         string
 	seqid          int
 	arg            interface{}
 	res            interface{}
 	errorUnwrapper ErrorUnwrapper
 	profiler       Profiler
+	ctx            context.Context
```

Comment:
Having ctx as a struct field is frowned upon, so perhaps have a comment explaining that call is a short-lived struct just meant for passing arguments across a channel, so it's ok. (And move ctx to be the first field, to mirror the convention that ctx is always the first function argument.)